core/modules/nfv_ctrl.h

```diff
@@ -37,14 +38,19 @@ class NFVCtrl final : public Module {
 
  private:
   // All available per-core packet queues in a cluster
+  std::map<uint16_t, uint16_t> longTermOptimization(std::vector<double> flow_rate_per_bucket);
+  std::map<uint16_t, uint16_t> findMoves(double flow_rate_per_cpu[], std::vector<uint16_t> to_be_moved, std::vector<double> flow_rate_per_bucket);
   std::vector<WorkerCore> cpu_cores_;
   uint64_t long_epoch_update_period_;
   uint64_t long_epoch_last_update_time_;
-  int total_core_count_ = 0;
+  uint16_t total_core_count_ = 0;
   uint64_t slo_p50_ = 200000000; //200ms
 
   // The lock for maintaining a pool of software queues
   mutable std::shared_mutex sw_q_mtx_;
+  PMDPort *port_;
+  std::map<uint16_t, std::vector<uint16_t>> core_bucket_mapping_;
+  std::map<uint64_t, uint64_t> threshold_;
```

Comment:
|threshold_| is unclear.
Either change the name to indicate what it does, or add a comment.